### PR TITLE
Remove image metrics integration tests

### PR DESCRIPTION
**Pull request by zakkak**
As part of https://github.com/quarkusio/quarkus/issues/40076 is now implemented and we are gathering the necessary stats in a database there is no longer the need for these tests to exist.

Note that automated alerting in case of regressions is still not setup but we now have a way to go back in time and see which (few) patches resulted in the regression, even if we don't spot it immediately.

### CODE REVIEW

**Review by yrodiere — APPROVED** [review]
> Note that automated alerting in case of regressions is still not setup but we now have a way to go back in time and see which (few) patches resulted in the regression, even if we don't spot it immediately.

Thanks for working on that!

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/NativeBuildOutputExtension.java [hunk 1]
```diff
@@ -1,147 +0,0 @@
...
-
-/**
- * This is a general utility to assert via
- * unit testing how many classes, methods, objects etc. have been included in a native-image.
- * <p>
- * For detailed information and explanations on the build output, visit
- * <a href="https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md">the upstream GraalVM
- * documentation</a>.
- */
-public class NativeBuildOutputExtension implements BeforeAllCallback {
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/NativeBuildOutputExtension.java:32]: That would be the safest option, yes. Up to you.

**Review by yrodiere — APPROVED** [review]

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/NativeBuildOutputExtension.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zakkak) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/NativeBuildOutputExtension.java:32]: Kept and added @Deprecated will merge once CI is happy.

**Review by gsmet — APPROVED** [review]

**Comment by zakkak** [comment]
CI failures seem unrelated

**Pull request merged**